### PR TITLE
LVLAVS_675_361_Integration_tests

### PR DESCRIPTION
**Pull request by ValeryGutor**
Was created configuration for integration tests, unit and integration tests for dao, service, repositories layers with all needed files, resources and utils.

### CODE REVIEW

File: src/test/java/ua/softserveinc/tc/util/UserUtils.java [hunk 1]
```diff
@@ -8,33 +8,55 @@
...
+        User user5 = new User();
+        user5.setId(1L);
+        user5.setFirstName("Alan");
+        user5.setLastName("Bom");
+        user5.setEmail("user@softserveinc.com");
+        user5.setPassword("$2a$08$6fjMaYthaRD9XpOQ7V652.N/pRpmOqdrRMU5b1otTRveK0T3pYa02");
+        user5.setRole(Role.USER);
+        user5.setPhoneNumber("+380987654321");
+        user5.setConfirmed(true);
+        user5.setActive(true);
```
Comment (andryspo) [inline src/test/java/ua/softserveinc/tc/util/UserUtils.java]: maybe create function for setting user parameters, it seems like a lot of code duplications

File: src/test/java/ua/softserveinc/tc/services/unit/CalendarServiceTest.java [hunk 2]
```diff
@@ -0,0 +1,77 @@
...
+    public void testFindEventByRoomId(List<Event> listOfEvent, Long roomId) {
+        when(roomService.findEntityById(roomId)).thenReturn(new Room());
+        when(roomService.findEntityById(roomId).getEvents()).thenReturn(listOfEvent);
+
+        verify(roomService, times(0)).findEntityById(roomId);
+    }
+
+    private static final Object[] getEvents() {
+        return $(
+                $(Arrays.asList(new Event(), new Event(), new Event()), 0L)
```
Comment (andryspo) [inline src/test/java/ua/softserveinc/tc/services/unit/CalendarServiceTest.java]: just interesting what does symbol "$" do here?

File: src/test/java/ua/softserveinc/tc/hibernate.test.properties [hunk 3]
```diff
@@ -0,0 +1,12 @@
...
+jdbc.username = root
+jdbc.password = melancholiya20
+hibernate.dialect = org.hibernate.dialect.MySQLDialect
+hibernate.show_sql = true
+hibernate.format_sql = true
+hibernate.hbm2ddl.auto = create
+hibernate.enable_lazy_load_no_trans = true
+hibernate.search.default.directory_provider = ram
+hibernate.search.default.indexBase = com.acme.hibernate.CustomDirectoryProvider
+hibernate.search.indexing_strategy = event
```
Comment (andryspo) [inline src/test/java/ua/softserveinc/tc/hibernate.test.properties]: please don`t forget about empty lines

File: src/test/java/ua/softserveinc/tc/repo/BookingRepositoryTestIT.java [hunk 4]
```diff
@@ -0,0 +1,83 @@
...
+    @Test
+    public void testFindByBookingStateWhenThereIsMultipleBookings() {
+
+        List<Booking> active_bookings = bookingRepository.findByBookingState(BookingState.ACTIVE);
+        Assert.assertEquals(BookingMessages.COUNT_BY_ROOM_AND_BOOKING_STATE, 2, active_bookings.size());
+
+        List<Booking> blocked_bookings = bookingRepository.findByBookingState(BookingState.BOOKED);
+        Assert.assertEquals(BookingMessages.COUNT_BY_ROOM_AND_BOOKING_STATE, 0, blocked_bookings.size());
+    }
+
```
Comment (thrytsko-amplify) [inline src/test/java/ua/softserveinc/tc/repo/BookingRepositoryTestIT.java]: extra line

File: src/test/java/ua/softserveinc/tc/util/RateUtils.java [hunk 5]
```diff
@@ -0,0 +1,18 @@
...
+
+import java.util.Arrays;
+import java.util.List;
+
+public class RateUtils {
+
+    public static List<Rate> getListOfRates() {
+        Rate rate1 = new Rate();
+        Rate rate2 = new Rate();
+        Rate rate3 = new Rate();
```
Comment (thrytsko-amplify) [inline src/test/java/ua/softserveinc/tc/util/RateUtils.java]: Unneeded variables

File: src/test/java/ua/softserveinc/tc/util/UserUtils.java (hunk 1 above)
Comment (ValeryGutor) [inline src/test/java/ua/softserveinc/tc/util/UserUtils.java]: Oh, great idea) I will rewrite it)

File: src/test/java/ua/softserveinc/tc/services/unit/CalendarServiceTest.java (hunk 2 above)
Comment (ValeryGutor) [inline src/test/java/ua/softserveinc/tc/services/unit/CalendarServiceTest.java]: This is used for JUnitParams

File: src/test/java/ua/softserveinc/tc/hibernate.test.properties (hunk 3 above)
Comment (ValeryGutor) [inline src/test/java/ua/softserveinc/tc/hibernate.test.properties]: Ok, miss that one..

File: src/test/java/ua/softserveinc/tc/repo/BookingRepositoryTestIT.java (hunk 4 above)
Comment (ValeryGutor) [inline src/test/java/ua/softserveinc/tc/repo/BookingRepositoryTestIT.java]: Ok)

File: src/test/java/ua/softserveinc/tc/util/RateUtils.java (hunk 5 above)
Comment (ValeryGutor) [inline src/test/java/ua/softserveinc/tc/util/RateUtils.java]: You're right, I shouldn't create this variables instead I should mock this, but I try to use JUnitParams  and miss this two classes (RateDaoTest.java, EventDaoTest.java) when refactoring. Test method uses this variables just to pass to the method. 
I will rewrite it.

**Pull request merged**